aten/src/ATen/native/mkldnn/xpu/Blas.cpp

```diff
@@ -92,10 +92,16 @@ Tensor& addmm_out(
           1.f, alpha.to<float>(), 0.f, attr.kind_with_linear);
     }
   } else {
+    // We use post_binary here for adding self matrix.
+    // To avoid wrong write, here we clone self for inplace case.
     if (alpha.to<float>() == 1.f && beta_ == 1.f) {
-      bias = self;
+      bias = is_inplace ? self.clone() : self;
     } else {
-      Tensor binary = self.dim() == 1 ? self.unsqueeze(0) : self;
+      Tensor binary;
+      if (is_inplace)
+        binary = self.dim() == 1 ? self.unsqueeze(0).clone() : self.clone();
+      else
+        binary = self.dim() == 1 ? self.unsqueeze(0) : self;
```

Comment:
It would be better to add comments to describe why `unsqueeze` is required. Or add a utility function for ndim alignment.

Reply:
`unsqueeze` here is used to handle `mv` related operators. The comments have been updated.